src/NET8SnapStart/DeleteProduct/Function.cs

```diff
@@ -0,0 +1,139 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Core;
+using Shared;
+using Shared.DataAccess;
+using Shared.Models;
+
+[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]
+
+namespace DeleteProduct
+{
+    public class Function
+    {
+        private readonly ProductsDAO dataAccess;
+        public Function()
+        {
+            this.dataAccess = new DynamoDbProducts();
+            Amazon.Lambda.Core.SnapshotRestore.RegisterBeforeSnapshot(StuffToRunBeforeSnapshotIsTaken);
+            Amazon.Lambda.Core.SnapshotRestore.RegisterAfterRestore(StuffToRunAfterRestore);
+        }
+
+        private ValueTask StuffToRunAfterRestore()
+        {           
+            /*
+            We don't use this method in this code, but keeping it here for others to use as a sample.
+            Some reason that you may need to run code after restore:
+            * Reestablish connections
+            * Re-seed uniqueness such as recreating a pseudo-random number generator or creating a new GUID
+            * Refresh configuration, such as config that is downloaded from the network
+            */
+            return ValueTask.CompletedTask;
+        }
+
+        private async ValueTask StuffToRunBeforeSnapshotIsTaken()
+        {           
+            /*
+            We use this method to warm up the JIT compiler to improve post-restore performance, but it can also 
+            be used to flush logs, or close connections, or general cleanup before the snapshot is taken.
+            */
+            
+            // Run this 100 times to make sure JIT is fully warmed up, make sure to stay under the 10 second limit
```

Comment:
We should make it clear that 100 is a magic number. People might wonder, why 100?

Reply:
Also feels like a lot of http calls as each call comes with some $ value

Reply:
Okay. I'll add a comment explaining. We don't care about optimizing for cost here, but I'll mention that in the comment.